queries/eager_load.go

```diff
@@ -190,6 +190,21 @@ func (l loadRelationshipState) callLoadFunction(depth int, loadingFrom reflect.V
 	return nil
 }
 
+// firstNonNilValue returns the first non nil value of input. If all are nil, return the value with index 0.
```

Comment:
I dont understand this logic. If I have Status.ChangedBy.Emails, what is the val in this case? what is 0-length?I just want to better understand the logic for this.